torchio/data/queue.py

```diff
@@ -146,15 +147,39 @@ def __init__(
         self.shuffle_subjects = shuffle_subjects
         self.shuffle_patches = shuffle_patches
         self.samples_per_volume = samples_per_volume
+        if type(samples_per_volume) == int:
+            self.samples_per_volume = [samples_per_volume for _ in range(
+                self.num_subjects)]
+        assert type(self.samples_per_volume) == list, ('`samples_per_volume`'
```

Comment:
Please relax this to `Sequence` (with `isinstance`) and raise a `ValueError`, instead of using `assert`.

Reply:
I can do this. Why not raising TypeError?

Reply:
You're right, `TypeError` would be more appropriate.